### PR TITLE
[ENH,RF] Multiecho support redux & --files usage changes

### LINKED ISSUE #158: Question about different echo time for one serie
Hi, 

I am curious about the behavior of heudiconv and your position concerning the following situation:

I have a situation where I got two different echo times for a single DICOM serie. This happened with the siemens fieldmap sequence fm2d2r which produces two series : one whit the two magnitude scans - one for each echo time - and one with the phase difference.

First I tried to use the echo time to get my two nifti files with the proper magnitude1 and magnitude2 key but it was not working as heudiconv was offering me only one echo time for the whole serie.

I have been able to deal with this in order to get my dataset in the BIDS format. However I was wondering how heudiconv deals with this kind of case ? For example in the dicominfo.tsv, I see that there is only one line for this serie, showing me only one echo time.

I was wondering if it could be better to have two lines, one for each echo time. Or one line for each parameter diverging in the serie in a more general way.

And thank you for this great tool!

Best,

Grégory Bliault

### PR DESCRIPTION
**Pull request by mgxd**
Closes #158, #162, #197, #203, #255

Original PR started in #259, with thanks to @pvelasco

This PR contains initial support and testing for multi-echo sequences (in BIDS structure).

For multi-echo functional data, it generates a single _events.tsv file (following BIDS spec 1.1.0).
It addresses issues #162 , #203 , #255 and maybe some others.
It has been tested on CMRR multi-band multi-echo EPI and MGH multi-echo MPRAGE data from a Prisma (VE11C), in both cases with magnitude and phase images.

* adds new heuristic `bids-ME.py` for multi-echo testing
* `tests/utils.py` refactor to allow re-use across different datalad installations / heudiconv runs 

Additionally, this also addresses #197 - allowing the usage of `--files` without requiring the converting heuristic to have `infotoids` method. This only works iff 1 `--subject` is provided.

### CODE REVIEW

**Comment by mgxd** [comment]
no objections @yarikoptic ?

File: heudiconv/convert.py [hunk 1]
```diff
@@ -471,16 +474,119 @@ def save_converted_files(res, item_dicoms, bids, outtype, prefix, outname_bids,
...
+                        this_prefix_basename = this_prefix_basename.replace(
+                            imgtype, "_echo-%d%s" % (echo_number, imgtype)
+                        )
+                        break
+
+            # For Scout runs with multiple NIfTI images per run:
+            if bids and 'scout' in this_prefix_basename.lower():
+                # in some cases (more than one slice slab), there are several
+                #   NIfTI images in the scout run, so distinguish them with "_acq-"
+                spt = this_prefix_basename.split('_acq-Scout', 1)
```
[the remark this answers is not in the document]
Comment (mgxd) [reply heudiconv/convert.py]: i think we can just exclude this scout section until we find a more flexible way to approach it.

**Comment by mgxd** [comment]
pinging this - @yarikoptic any other thoughts?

**Comment by yarikoptic** [comment]
not really besides a feeling that I am not exactly sure what is going on ;) but that is Ok I guess, time will show -- let's just move on and deal with it later ;)

**Pull request merged**